src/QlikConnectorPSExecute/UserInteractive.cs

```diff
@@ -0,0 +1,346 @@
+            sid = Marshal.AllocHGlobal(sidSize);
+
+            if (!LookupAccountName(string.Empty, accountName, sid, ref sidSize, domainName, ref nameSize, ref accountType))
+            {
+                winErrorCode = GetLastError();
+                errorMessage = ($"LookupAccountName failed: {winErrorCode}");
+                throw new Win32Exception((int)winErrorCode, errorMessage);
+            }
+            else
+            {
+                var systemName = new LSA_UNICODE_STRING();
+
+                var policyHandle = IntPtr.Zero;
+                var userRightsPtr = IntPtr.Zero;
+                int countOfRights = 0;
+
+                var objectAttributes = CreateLSAObject();
+
+                uint policyStatus = LsaOpenPolicy(ref systemName, ref objectAttributes, Access, out policyHandle);
+                winErrorCode = LsaNtStatusToWinError(policyStatus);
+
+                if (winErrorCode != 0)
+                {
+                    errorMessage = ($"OpenPolicy failed: {winErrorCode}.");
+                    throw new Win32Exception((int)winErrorCode, errorMessage);
+                }
+                else
+                {
+                    try
+                    {
+                        uint result = LsaEnumerateAccountRights(policyHandle, sid, out userRightsPtr, out countOfRights);
+                        winErrorCode = LsaNtStatusToWinError(result);
+                        if (winErrorCode != 0)
+                        {
+                            if (winErrorCode == 2)
+                                return new List<string>();
+
+                            errorMessage = string.Format("LsaEnumerateAccountRights failed: {0}", winErrorCode);
+                            throw new Win32Exception((int)winErrorCode, errorMessage);
+                        }
+
+                        var newPtr = IntPtr.Zero;
+                        if (IntPtr.Size == 8)
+                            newPtr = new IntPtr(userRightsPtr.ToInt64());
+                        else
+                            newPtr = new IntPtr(userRightsPtr.ToInt32());
+
+                        LSA_UNICODE_STRING userRight;
+
+                        int ptr = 0;
+                        for (int i = 0; i < countOfRights; i++)
+                        {
+                            userRight = (LSA_UNICODE_STRING)Marshal.PtrToStructure(newPtr, typeof(LSA_UNICODE_STRING));
+                            var userRightStr = Marshal.PtrToStringAuto(userRight.Buffer);
+                            rights.Add(userRightStr);
+                            ptr += Marshal.SizeOf(userRight);
+                        }
+                    }
+                    finally
+                    {
+                        LsaClose(policyHandle);
+                    }
+                }
+                FreeSid(sid);
+            }
+            return rights;
+        }
+
+        public void SetRight(string accountName, string privilegeName, bool remove)
```

Comment:
don't repeat yourself, you can make one private Method out of Get & Set Right and make only an if around the get set lines. Call that private from the public Get / Set Method